src/Symfony/Component/Cache/Traits/MemcachedTrait.php

```diff
@@ -198,7 +198,12 @@ protected function doSave(array $values, $lifetime)
             $lifetime += time();
         }
 
-        return $this->checkResultCode($this->getClient()->setMulti($values, $lifetime));
+        $encodedValues = array();
+        foreach ($values as $key => $value) {
+            $encodedValues[rawurlencode($key)] = $value;
+        }
+
+        return $this->checkResultCode($this->getClient()->setMulti($encodedValues, $lifetime));
```

Comment:
this is not decoding keys in the returned value. Can it have keys in it ?

Reply:
\Memcached::setMulti returns bool. Did I missed something else?